### PR TITLE
🐛 Fix permission error for init of hosted instances

### PR DESCRIPTION
**Pull request by Koncopd**
This fixes the permission error and also circumvents the need for CreateBucket permission for Locker.

### CODE REVIEW

File: lamindb_setup/dev/_settings_storage.py [hunk 1]
```diff
@@ -262,7 +262,7 @@ def type(self) -> str:
         import fsspec
 
         convert = {"file": "local"}
-        protocol = fsspec.utils.get_protocol(str(self.root))
+        protocol = fsspec.utils.get_protocol(self.root_as_str)
```
Comment (Koncopd) [inline lamindb_setup/dev/_settings_storage.py:265]: It is called here

[lamindb-setup/lamindb_setup/dev/_hub_core.py](https://github.com/laminlabs/lamindb-setup/blob/3bff9bc24da7a9df7cdd00a84fe1599a5ce900ab/lamindb_setup/dev/_hub_core.py#L76)

Line 76 in [3bff9bc](https://github.com/laminlabs/lamindb-setup/commit/3bff9bc24da7a9df7cdd00a84fe1599a5ce900ab)

 type=ssettings.type, 
initializing the root and getting the credentials before the instance is registered in the hub.

**Comment by Koncopd** [comment]
This has problem with `set.storage`, i will debug later.

**Comment by falexwolf** [comment]
Thanks a ton for this, Sergei!

**Comment by Koncopd** [comment]
@falexwolf the only problem left i believe is linting in build / prod-only

**Comment by falexwolf** [comment]
Ok, I'll fix linting!

**Pull request merged**